src/Chapter1.hs

```diff
@@ -542,7 +547,16 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+qsort :: Ord a => [a] -> [a]
+qsort [] = []
+qsort [x] = [x]
+qsort (x:xs) = smol ++ (x : lorg)
+    where smol = qsort [z | z <- xs, z <= x]
+          lorg = qsort [z | z <- xs, z > x]
+
+mid :: Ord a => a -> a -> a -> a
+mid x y z = (!! 1) (qsort [x,y,z])
```

Comment:
Very creative solution! :smile: Looks like you already mastered the List topic in Chapter one! Though, of course, reusing the standard `sort` function will result in a shorter code :slightly_smiling_face:

Reply:
Wanted to save a dependency ;)